test/cloudtest/test_storaged.py

```diff
@@ -48,6 +48,49 @@ def test_storaged_creation(mz: MaterializeApplication) -> None:
         wait(condition="condition=Ready", resource=storaged)
 
 
+def test_storaged_resizing(mz: MaterializeApplication) -> None:
+    """Test that resizing a given source causes the storaged to be replaced"""
+    mz.testdrive.run_string(
+        dedent(
+            """
+            $ kafka-create-topic topic=test
+
+            $ kafka-ingest format=bytes topic=test
+            ABC
+
+            > CREATE SOURCE source1
+              FROM KAFKA BROKER '${testdrive.kafka-addr}'
+              TOPIC 'testdrive-test-${testdrive.seed}'
+              FORMAT BYTES
+              ENVELOPE NONE;
+            """
+        )
+    )
+    id = mz.environmentd.sql_query(f"SELECT id FROM mz_sources WHERE name = 'source1'")[
+        0
+    ][0]
+    assert id is not None
+    storaged = f"pod/storage-{id}-0"
+
+    wait(condition="condition=Ready", resource=storaged)
+    unaltered = mz.kubectl("get", storaged, "-o", "jsonpath=jsonpath='{.status.startTime}'")
+
+    mz.testdrive.run_string(
+        dedent(
+            """
+            > ALTER SOURCE source1
+              SET (SIZE '16');
+            """
+        ),
+        no_reset=True,
+    )
+
+    wait(condition="condition=Ready", resource=storaged)
+    altered = mz.kubectl("get", storaged, "-o", "jsonpath=jsonpath='{.status.startTime}'")
+
+    assert unaltered != altered, "Altering a source should restart the backing container"
+
+
```

Comment:
Please drop the source at the end of the test and then wait for condition="delete" to both clean up the cluster and confirm that altering has not prevented `DROP` from working properly.